forge-game/src/main/java/forge/game/card/CardFactory.java

```diff
@@ -371,6 +363,37 @@ private static void buildPlaneAbilities(Card card) {
         card.addSpellAbility(planarRoll);
     }
 
+    private static void buildBattleAbilities(Card card) {
+        if (!card.isBattle()) {
+            return;
+        }
+        // # The following commands should be pulled out into the codebase
+        //K:etbCounter:DEFENSE:3
+
+        if (card.getType().hasSubtype("Siege")) {
+            StringBuilder triggerSB = new StringBuilder();
+            triggerSB.append("Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Static$ True | ");
```

Comment:
"As enters the battlefield" should probably be an Replacement Effect

Reply:
Yea definitely.